### ISSUE TITLE
`len_without_is_empty` on type alias when both `len` and `is_empty` are implemented

### ISSUE THREAD

**Issue by davidcole1340** [rust-lang/rust-clippy#7702]
I tried this code:

```rust
pub struct Test;
pub type ExampleAlias = Test;

impl ExampleAlias {
    pub fn len(&self) -> usize {
        5
    }
    
    pub fn is_empty(&self) -> bool {
        false
    }
}
```

I expected to see this happen: No clippy error.

Instead, this happened:
```
warning: type `ExampleAlias` has a public `len` method, but no `is_empty` method
 --> src/main.rs:5:5
  |
5 |     pub fn len(&self) -> usize {
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(clippy::len_without_is_empty)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#len_without_is_empty
```

[Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=43514677e3a954b1f640cf8e21e7e6cc)

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.57.0-nightly (aa8f2d432 2021-09-18)
binary: rustc
commit-hash: aa8f2d432b23575929a48f87b8746f41ba723318
commit-date: 2021-09-18
host: x86_64-unknown-linux-gnu
release: 1.57.0-nightly
LLVM version: 13.0.0
```

Have also tried on `rustc 1.55.0` stable.

**Comment by y21** [comment]
Closing as the lint no longer triggers here (probably got fixed by #11452 - one of the added testcases looks identical at least)
